PAAS-8027 Add failed pods rescheduler (#49)
ankilosaurus/kube_remediator: README.md

Deletes `Pod`s with label `kube-remediator/OldPodDeleter=true` older than 24h

### [Failed Pods Rescheduler](pkg/remediator/failedpodsrescheduler.go)

Reschedules `Failed` `Pods` by deleting them, since they are not automatically cleaned up.

- Listens to Pod update events and does a Pod list

> - Looks for containers in Failed status with reason `OutOfCpu` or `OutofMemory`.

Reviewer: ```suggestion
- Finds pods in Failed status with reason `OutOfCpu`, `OutofMemory`, or `UnexpectedAdmissionError`.
```

> - Finds pods in Failed status with reason `OutOfCpu`, `OutofMemory`, or `UnexpectedAdmissionError`.

Reviewer: ```suggestion
- Finds pods in Failed status with reason `OutOfCpu` or `OutofMemory`
```

- Ignores Pods without `ownerReferences` (Avoid deleting something which does not come back)
- Ignores Pods for Jobs because they can be automatically cleaned up.

> 

Reviewer: say something about 5 minutes

Author: Added

### Unbound PersistentVolumeClaim cleaner TODO

> Deletes `PersistentVolumeClaim` left behind by deleted `StatefulSet`, that are not automatically cleaned up up otherwise

Reviewer: ```suggestion
Deletes `PersistentVolumeClaim` left behind by deleted `StatefulSet`, that are not automatically cleaned up otherwise
```
